test-data/unit/check-inference.test

```diff
@@ -380,6 +380,8 @@ class Nums(Iterable[int]):
     def __iter__(self): pass
     def __next__(self): pass
 a, b = Nums()
+reveal_type(a)  # N: Revealed type is "builtins.int"
```

Comment:
The other option here is to infer `Any`, because of `def __iter__(self): pass`

Reply:
Yup, but as discussed in the issue, two long-standing tests fail if I make that change, so I'd rather not make that behaviour change in this PR 👍